R/legend.grob.R

```diff
@@ -80,154 +80,163 @@ legend.grob <- function(
-						
-						# need padding to line up with key style legends
-						color.key.grob$framevp$x = unit(0,'npc') + unit(1.69,'points');
-						
-						# set sizes
-						color.key.grob$framevp$height <- unit(legendi[['height']], 'lines');
-						color.key.grob$framevp$width <- unit(legendi[['width']], 'lines');
-
-						if(!is.null(legendi[['pos.x']])) {
-							color.key.grob$framevp$x <- unit(legendi[['pos.x']],'npc');
-							}
-
-						if(!is.null(legendi[['pos.y']])) {
-							color.key.grob$framevp$y <- unit(legendi[['pos.y']],'npc');
-							}
-
-						# Add the legend to the frame
-						legend.grob.final <- packGrob(
-							frame = legend.grob.final,
-							grob = color.key.grob,
-							row = 3 * (legend.row - 1) + 2,
-							col = 2 * (legend.col - 1) + 1,
-							height = max(unit(legendi[['height']], 'lines'),
-									legend.grob.final$framevp$layout$heights[3 * (legend.row - 1) + 1]),
-							force.height = TRUE,
-							);
-						}
-					else {
-						legend.key <- list(
-							just = c('left', 'top'),
-							between = 0.5,
-							rep = FALSE,
-							rectangles = list(
-								col = legendi[['colours']],
-								size = if (is.null(legendi[['size']])) { size } else { legendi[['size']] },
-								height = 1,
-								border = legendi[['border']]
-								),
-							text = list(
-								legendi[['labels']],
-								cex = label.cex,
-								fontfamily = font.family
-								)
-							);
-
-						# Add the legend to the frame
-						legend.grob.final <- packGrob(
-							frame = legend.grob.final,
-							grob = draw.key(key = legend.key, draw = FALSE),
-							row = 3 * (legend.row - 1) + 2,
-							col = 2 * (legend.col - 1) + 1
-							);
-						}
+			# Figure out where this legend should go in the layout (ordered row-wise)
+			legend.row <- if (i %% layout[1] == 0) { i / layout[1] } else { floor(i / layout[1]) + 1 }
+			legend.col <- if (i %% layout[1] == 0) { layout[1] } else { i %% layout[1] }
+
+			# Create the title of this legend
+			if (!is.null(legendi[['title']])) {
+
+				# Draw a grob representing the title
+				title.x.coord <- 0.5;
+
+				if (title.just == 'left') {
+					title.x.coord <- 0;
 					}
-				);
+				else if (title.just == 'right') {
+					title.x.coord <- 1;
+					}
+
+				title.grob <- textGrob(
+					label = legendi[['title']],
+					just = c(title.just, 'top'),
+					x = title.x.coord,
+					y = 1,
+					gp = gpar(
+						cex = title.cex,
+						fontface = title.fontface,
+						fontfamily = font.family,
+						fill = 'black'
+						)
+					);
+
+				# Get the height of this grob so we can add white space around it
+				title.grob.height <- convertHeight(
+					grobHeight(title.grob),
+					unitTo = 'lines',
+					valueOnly = TRUE
+					);
+
+
+				# Add the title to the frame
+				legend.grob.final <- packGrob(
+					frame = legend.grob.final,
+					grob = title.grob,
+					row = 3 * (legend.row - 1) + 1,
+					col = 2 * (legend.col - 1) + 1,
```

Comment:
needs comments explaining why 3* and 2* on lines 125 and 126 respectively

Reply:
This behaviour comes from the [initial commit](https://github.com/uclahs-cds/public-R-BoutrosLab-plotting-general/commit/d31b7aa393bc8d0712bbec8d7a77dffe4e703f67) (moving the code from SVN).